enhancements/update/upgrades-blocking-on-ack.md

```diff
@@ -0,0 +1,138 @@
+Each key represents a specific upgrade blocking condition that must be acknowledged before minor level upgrades are allowed.
+An administrator acknowledges a condition by adding the corresponding key with a value of `true` to the configmap.
+Values other than `true` will be ignored and therefore not clear the upgrade block.
+The keys will be named to make it explicitly clear what the administrator is acknowledging and thereby agreeing to.
+CVO will set `Upgradeable=false` which in turn blocks cluster upgrades to the next minor release until it finds each of the predefined keys in the configmap with a value of `true`.
+For each key blocking an upgrade, CVO will generate a detailed message pointing to relevant OCP documentation explaining what the administrator is expected to do before acknowledging task completion, i.e. entering the corresponding key into the configmap with a value of `true`.
+
+The new configmap `upgrade-ack` will be installed as part of the release in the namespace `openshift-cluster-version`.
+It will initially have no keys since the administrator is expected to add the relevant key as part of acknowledgement.
+
+### User Stories
+
+Clusters upgraded to 4.9 will no longer have access to these [APIs][k8s-1.22-removed-apis].
+This new upgrade blocking mechanism will be used in an attempt to require administrators to read detailed documentation regarding the removed APIs and the possible impact.
+This OCP documentation will list which APIs have been removed and how an administrator checks their cluster for use of those APIs.
+
+4.8.z versions that are upgradeable to 4.9 will contain this enhancement and have the configmap `openshift-cluster-version/upgrade-ack`.
+CVO will set `Upgradeable=false` which in turn blocks cluster upgrades to 4.9 until the administrator has added the key/value pair `ack-4.9-api-removal-upgrade-approved: "true"` to the configmap.
+Until this is done the `Conditions` console page will display an `Upgradeable` condition message directing the administrator to the relevant OCP documentation.
+
+If the administrator wants to upgrade to 4.9 without performing the acknowledgment `force` can be used.
+
+### Implementation Details/Notes/Constraints
+
+### Risks and Mitigations
+
+* A common risk is human error resulting in an incorrect key/value pair being placed in the configmap.
+This type of error will not change `Upgradeable=false` and therefore the administrator will know a mistake has been made since the `Upgradeable` condition message will continue to be displayed.
+CVO will log an error whenever an unrecognized key is found in the configmap.
+* Another risk is that the configmap gets removed.
+This will result in `Upgradeable=false` and the standard `Upgradeable` condition message will be displayed with the additional message that the configmap cannot be found and should be created. (OPTION: Should CVO create the configmap?)
```

Comment:
This seems fine if the individual acks are enumerated in code, if they're a shipped manifest then we'd need them to be create-only and one per ACK?

Reply:
Perhaps another reason to have both a "keys config" configmap and an ack configmap with the above only applying to the latter.

Reply:
I think we need to clearly mention that option. I am in favor of having two config maps. One gets created by CVO for a situation where we need approval from admin to move forward with update, 2nd Admin creates to give approval. This way admin can delete and recreate if required without messing things up.